tests/sentry/backup/test_correctness.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import annotations
+
+import os
+from difflib import unified_diff
+from pathlib import Path
+from typing import NewType
+
+from click.testing import CliRunner
+from freezegun import freeze_time
+from pydantic import FilePath, PositiveInt
+
+from sentry.db.postgres.roles import in_test_psql_role_override
+from sentry.runner.commands.backup import export, import_
+from sentry.utils import json
+from sentry.utils.json import JSONData, JSONEncoder, better_default_encoder
+from sentry.utils.pytest.fixtures import django_db_all
+
+ComparatorName = NewType("ComparatorName", str)
+ModelName = NewType("ModelName", str)
+
+
+# TODO(team-ospo/#155): Figure out if we are going to use `pk` as part of the identifier, or some other kind of sequence number internal to the JSON export instead.
+class InstanceID:
+    """Every entry in the generated backup JSON file should have a unique model+pk combination, which serves as its identifier."""
+
+    def __init__(self, model: ModelName, pk: PositiveInt):
+        self.model = model
+        self.pk = pk
+
+    def __eq__(self, other):
+        return (
+            hasattr(other, "model")
+            and self.model == other.model
+            and hasattr(other, "pk")
+            and self.pk == other.pk
+        )
+
+    def __hash__(self):
+        return hash((self.model, self.pk))
+
+    def print(self):
+        return f'InstanceID(model: "{self.model}", pk: {self.pk})'
+
+
+class ComparatorFinding:
```

Comment:
I would probably replace this with `typing.NamedTuple` directly:

```python
class ComparatorFinding:
    name: str
    on: InstanceId
    reason: str | None = ""
```

although now that I look at it -- I don't think `reason` is nullable -- so it can just be `str`?

probably the same for the other plain-old-data classes.  then you don't need to implement repr either as it'll come for free

Reply:
Done.